smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaAdmin.java

```diff
@@ -0,0 +1,20 @@
+package io.smallrye.reactive.messaging.kafka;
+
+import java.util.Set;
+
+import org.apache.kafka.clients.admin.Admin;
+
+import io.smallrye.mutiny.Uni;
+
+/**
+ * Internal interface for Kafka admin client.
+ * To complete with remaining {@link org.apache.kafka.clients.admin.Admin} method wrappers, if decided to expose it externally.
+ */
+public interface KafkaAdmin {
+
+    Uni<Set<String>> listTopics();
+
+    Admin unwrap();
+
+    void closeAndAwait();
```

Comment:
I would return the Uni<Void> instead, as the listTopics is returning a Uni.

Reply:
Just got the problem with the Uni approach. Let's keep closeAndAwait.

Reply:
We can wrap the AdminClient#close call inside a Uni but if we choose not to have a worker thread for this client it'll still block the caller thread.

Reply:
Yes, but unnecessary I believe.